### ISSUE TITLE
connect_backend: accessing a non existent slab entry

### ISSUE THREAD

**Issue by BlackYoup** [sozu-proxy/sozu#501]
This panic occured:
```
#0  0x00007f1ee301d996 in __strlen_sse2 () from /usr/x86_64-pc-linux-gnu/lib/libc.so.6
#1  0x00005563885903ff in std::sys_common::gnu::libbacktrace::resolve_symname ()
#2  0x000055638859ba6b in std::sys_common::backtrace::print ()
#3  0x000055638857d23c in std::panicking::default_hook::{{closure}} ()
#4  0x000055638857cf14 in std::panicking::default_hook ()
#5  0x000055638857d8b6 in std::panicking::rust_panic_with_hook ()
#6  0x000055638857d3fd in std::panicking::begin_panic_fmt ()
#7  0x000055638857d3c2 in rust_begin_unwind ()
#8  0x00005563885eb003 in core::panicking::panic_fmt ()
#9  0x00005563885f5da2 in core::option::expect_failed ()
#10 0x000055638829b488 in <core::option::Option<T>>::expect (msg=..., self=...) at /var/tmp/paludis/build/dev-lang-rust-1.27.1/work/rustc-1.27.1-src/src/libcore/option.rs:302
#11 <slab::Slab<T, I> as core::ops::index::Index<I>>::index (self=<optimized out>, index=...) at /var/cache/paludis/distfiles/cargo-home/registry/src/github.com-1ecc6299db9ec823/slab-0.3.0/src/lib.rs:277
#12 sozu_lib::network::proxy::Server::connect_to_backend (self=0x7ffdc429b250, token=...) at lib/src/network/proxy.rs:748
#13 0x000055638829c01e in sozu_lib::network::proxy::Server::ready (self=0x7ffdc429b250, token=..., events=...) at lib/src/network/proxy.rs:932
#14 0x00005563882944a7 in sozu_lib::network::proxy::Server::run (self=0x7ffdc429b250) at lib/src/network/proxy.rs:392
#15 0x000055638813e3fe in sozu::worker::begin_worker_process (fd=<optimized out>, scm=<optimized out>, configuration_state_fd=<optimized out>, id=<optimized out>, command_buffer_size=<optimized out>, max_command_buffer_size=<optimized out>) at bin/src/worker.rs:87
#16 0x00005563880afbd7 in sozu::cli::upgrade_worker::{{closure}} (worker_matches=<optimized out>) at bin/src/cli.rs:92
#17 <core::option::Option<T>>::and_then (self=..., f=...) at /var/tmp/paludis/build/dev-lang-rust-1.27.1/work/rustc-1.27.1-src/src/libcore/option.rs:612
#18 sozu::cli::upgrade_worker (matches=<optimized out>) at bin/src/cli.rs:84
#19 sozu::main () at bin/src/main.rs:57
```

It seems we are trying to access an entry in the slab that doesn't exist (anymore?):
https://github.com/sozu-proxy/sozu/blob/buffer-debug/lib/src/network/proxy.rs#L748
https://github.com/carllerche/slab/blob/v0.3.0/src/lib.rs#L273...L279

No much logs are available other than the stack trace. The log just before it happened was:
`2018-07-23T20:17:06Z 361464800169174 18825 WRK-04 ERROR      0760fedc-233c-4c53-92a7-740c801d7f9e    <REDACTED>         error connecting to backend, trying again`

**Comment by Geal** [comment]
this should be fixed by  29bbe1d

**Comment by BlackYoup** [comment]
Didn't reproduce since this commit
